hudi-common/src/main/java/org/apache/hudi/common/util/HoodieRecordUtils.java

```diff
@@ -75,14 +75,14 @@ public static HoodieRecordMerger loadRecordMerger(String mergerClass) {
    * Instantiate a given class with a record merge.
    */
   public static HoodieRecordMerger createRecordMerger(String basePath, EngineType engineType,
-                                                      List<String> mergerClassList, String recordMergerStrategy) {
-    if (mergerClassList.isEmpty() || HoodieTableMetadata.isMetadataTable(basePath)) {
+                                                      List<String> mergerClassList, Option<String> recordMergerStrategy) {
+    if (mergerClassList.isEmpty() || recordMergerStrategy.isEmpty() || HoodieTableMetadata.isMetadataTable(basePath)) {
```

Comment:
It would be better to use merge mode to determine the merger too, and this can be in the `RecordMergeConfig` class.

Reply:
If the merge mode is `CUSTOM`, use `HoodieAvroRecordMerger`.

Reply:
How is this method different from other `getRecordMerger` methods?  It would be good to unify all similar logic to a single method.

Reply:
To summarize, `HoodieRecordUtils#createRecordMerger` is the only method creating record merge implementation class instance.

Reply:
`HoodieTableConfig#inferCorrectMergingBehavior` is the single method that dictates the inference of merging configs.